tests/cmdline/commands/test_storage.py

```diff
@@ -0,0 +1,118 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for `verdi storage`."""
+import pytest
+
+from aiida.cmdline.commands import cmd_storage
+from aiida.common import exceptions
+
+
+@pytest.mark.usefixtures('clear_database_before_test')
+def tests_storage_info(aiida_localhost, run_cli_command):
+    """Test the ``verdi storage info`` command with the ``-statistics`` option."""
+    from aiida import orm
+    node = orm.Dict().store()
+
+    result = run_cli_command(cmd_storage.storage_info, options=['--statistics'])
+
+    assert aiida_localhost.label in result.output
+    assert node.node_type in result.output
+
+
+def tests_storage_migrate_force(run_cli_command):
+    """Test the ``verdi storage migrate`` command (with force option)."""
+    result = run_cli_command(cmd_storage.backend_migrate, options=['--force'])
+    assert result.output == ''
+
+
+def tests_storage_migrate_interactive(run_cli_command):
+    """Test the ``verdi storage migrate`` command (with interactive prompt)."""
+    from aiida.manage.manager import get_manager
+    profile = get_manager().get_profile()
+
+    result = run_cli_command(cmd_storage.backend_migrate, user_input='MIGRATE NOW')
+
+    assert 'warning' in result.output.casefold()
+    assert profile.name in result.output
+    assert 'migrate now' in result.output.casefold()
+    assert 'migration completed' in result.output.casefold()
+
+
+def tests_storage_migrate_running_daemon(run_cli_command, monkeypatch):
+    """Test that ``verdi storage migrate`` raises if the daemon is running."""
+    from aiida.engine.daemon.client import DaemonClient
+
+    def return_true():
+        return True
+
+    with monkeypatch.context() as mock_context:
+        mock_context.setattr(DaemonClient, 'is_daemon_running', return_true)
+        result = run_cli_command(cmd_storage.backend_migrate, raises=True)
+
+    assert 'daemon' in result.output.casefold()
+    assert 'running' in result.output.casefold()
+
+
+def tests_storage_migrate_cancel_prompt(run_cli_command, monkeypatch):
+    """Test that ``verdi storage migrate`` detects the cancelling of the interactive prompt."""
+    import click
+
+    def raise_click_abort(*args, **kwargs):
+        raise click.Abort()
+
+    with monkeypatch.context() as mock_context:
+        mock_context.setattr(click, 'prompt', raise_click_abort)
+        result = run_cli_command(cmd_storage.backend_migrate, raises=True)
```

Comment:
```suggestion
        monkeypatch.setattr(click, 'prompt', lambda x: raise click.Abort())
        result = run_cli_command(cmd_storage.backend_migrate, raises=True)
```

Reply:
Note I didn't test this, so you may have to massage the lambda definition a bit so it accepts arbitrary args and kwargs but it should work.

Reply:
Done. The lambda function apparently [doesn't play along nicely with raises](https://stackoverflow.com/questions/8294618/define-a-lambda-expression-that-raises-an-exception) so I had to use exec, but this is a controlled case where the typical vulnerability of exec does not apply. The line does look a bit clunky, with the need to import the exception and all that, so I left the old way commented out in case we want to use the other one for clarity. Anyways I will have to do at least one more update for the deprecation fixture.

Let me know which you prefer but don't resolve this one yet (hopefully it'll serve as a reminder that I need to remove the comments).

Reply:
(removed the commented version now btw)